apps/site/assets/ts/hooks/useRoute.ts

```diff
@@ -0,0 +1,56 @@
+import { groupBy, map } from "lodash";
+import useSWR from "swr";
+import { fetchJsonOrThrow } from "../helpers/fetch-json";
+import { RouteWithDirections, TypedRoutes } from "../stop/components/__stop";
+import { EnhancedRoute, Route } from "../__v3api";
+
+const fetchData = async (url: string): Promise<Route[]> =>
+  fetchJsonOrThrow(url);
+
+const typeToString = (type: number): String => {
+  switch (type) {
+    case 0: {
+    }
+    case 1: {
+      return "subway";
+    }
+    case 2: {
+      return "commuter-rail";
+    }
+    case 3: {
+      return "bus";
+    }
+    case 4: {
+    }
+    default: {
+      return "ferry";
+    }
+  }
+};
+
+const useRoutesByStop = (stopId: string): Route[] | undefined => {
+  const { data } = useSWR<Route[]>(`/api/routes/by-stop/${stopId}`, fetchData);
+  return data;
+};
+
+// TODO move this grouping to the backend
+const useTypedRoutesByStop = (stopId: string): TypedRoutes[] | undefined => {
+  const routes = useRoutesByStop(stopId);
+  const groupedRoutes = groupBy(routes, route => {
+    return typeToString(route.type);
+  });
+
+  return map(groupedRoutes, (dictRoutes, key) => {
+    return {
+      group_name: key,
+      routes: map(dictRoutes, r => {
+        return {
+          route: r as EnhancedRoute,
+          directions: []
+        } as RouteWithDirections;
+      })
+    } as TypedRoutes;
+  });
```

Comment:
This is a duplication of backend logic found in the [stop_controller](https://github.com/mbta/dotcom/blob/master/apps/site/lib/site_web/controllers/stop_controller.ex#L234)

If this logic was kept on the backend end, then the endpoint would be returning grouped routes.  When the stop page requires a list of routes, the page would either have to fetch a flat list from a different endpoint again, or flatten the grouped list.  The first option is not great as it requests the same data from the backend twice (just a slightly different form).  The flattening isn't great because we need to manipulate the data before being used.  It makes more sense to have the data in a standard state (Flat list), and then manipulate it to the form the p[age requires (grouped by type).
  
Also SWR has a built in request de-duplication which means when the page calls both `useRoutesByStop` and `useTypedRoutesByStop` if will only send 1 network request.

Another option is to rewrite all the downstream components to work with a flat list of routes instead of grouped routes.


Thoughts? Feedback? Other ideas?